File: lib/features/me/widgets/stats_tab.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';

import 'package:goworkbro/core/l10n/app_locale.dart';
import 'package:goworkbro/core/theme/app_theme.dart';
import 'package:goworkbro/providers/app_provider.dart';

/// Stats tab: lifetime and today's aggregated counters.
class StatsTab extends StatelessWidget {
  const StatsTab({super.key});

  @override
  Widget build(BuildContext context) {
    final provider = context.watch<AppProvider>();
    final theme = Theme.of(context);
    final s = S.of(context.read<AppLocaleProvider>().locale);
    return ListView(
      padding: const EdgeInsets.all(16),
      children: [
        Text(s.lifetimeStats, style: theme.textTheme.titleLarge),
        const SizedBox(height: 12),
        _buildStatCard(
          theme,
          title: s.totalFocus,
          value: _formatDuration(provider.lifetimeFocusSeconds),
          icon: Icons.all_inclusive,
          color: AppTheme.chartColors[0],
        ),
        const SizedBox(height: 12),
        _buildStatCard(
          theme,
          title: s.totalSessions,
          value: '${provider.lifetimeSessionCount}',
          icon: Icons.local_fire_department_outlined,
          color: AppTheme.chartColors[3],
        ),
        const SizedBox(height: 12),
        _buildStatCard(
          theme,
          title: s.totalTodos,
          value: '${provider.lifetimeTodosCompleted}',
          icon: Icons.task_alt,
          color: AppTheme.chartColors[2],
        ),
        const SizedBox(height: 12),
        _buildStatCard(
          theme,
          title: s.totalHabits,
          value: '${provider.lifetimeHabitsCompleted}',
          icon: Icons.repeat_rounded,
          color: AppTheme.chartColors[1],
        ),
        const SizedBox(height: 12),
        _buildStatCard(
          theme,
          title: s.usingSince,
          value: provider.firstUsedDate.split('T').first,
          icon: Icons.calendar_month_outlined,
          color: AppTheme.chartColors[4],
        ),
        const SizedBox(height: 24),
        Divider(color: theme.dividerColor),
        const SizedBox(height: 16),
        Text(s.today, style: theme.textTheme.titleLarge),
        const SizedBox(height: 12),
        _buildStatCard(
          theme,
          title: s.todayFocus,
          value: _formatDuration(provider.todayTotalFocusSeconds),
          icon: Icons.timer_outlined,
          color: AppTheme.chartColors[0],
        ),
        const SizedBox(height: 12),
        _buildStatCard(
          theme,
          title: s.pomodoroCount,
          value: s.count(provider.todaySessionCount),
          icon: Icons.local_fire_department_outlined,
          color: AppTheme.chartColors[3],
        ),
        const SizedBox(height: 12),
        _buildStatCard(
          theme,
          title: s.todoCompleted,
          value:
              '${provider.todos.where((t) => t.isCompleted).length} / ${provider.todos.length}',
          icon: Icons.check_circle_outline,
          color: AppTheme.chartColors[2],
        ),
        const SizedBox(height: 12),
        _buildStatCard(
          theme,
          title: s.habitCompleted,
          value:
              '${provider.habits.where((h) => h.isCompleted).length} / ${provider.habits.length}',
          icon: Icons.repeat_outlined,
          color: AppTheme.chartColors[1],
        ),
        const SizedBox(height: 12),
        _buildStatCard(
          theme,
          title: s.activeCountdowns,
          value: s.count(provider.countdowns.length),
          icon: Icons.hourglass_empty,
          color: AppTheme.chartColors[4],
        ),
      ],
    );
  }

  Widget _buildStatCard(
    ThemeData theme, {
    required String title,
    required String value,
    required IconData icon,
    required Color color,
  }) {
    return Card(
      child: Padding(
        padding: const EdgeInsets.all(20),
        child: Row(
          children: [
            Container(
              width: 48,
              height: 48,
              decoration: BoxDecoration(
                color: color.withValues(alpha: 0.12),
                borderRadius: BorderRadius.circular(14),
              ),
              child: Icon(icon, color: color, size: 24),
            ),
            const SizedBox(width: 16),
            Expanded(child: Text(title, style: theme.textTheme.bodyMedium)),
            Text(
              value,
              style: theme.textTheme.titleLarge?.copyWith(color: color),
            ),
          ],
        ),
      ),
    );
  }

  String _formatDuration(int seconds) {
    final h = seconds ~/ 3600;
    final m = (seconds % 3600) ~/ 60;
    if (h > 0) return '${h}h ${m}m';
    return '${m}m';
  }
}

File: app/src/main/java/com/azarai/goworkbro/ui/me/StatsTab.kt
package com.azarai.goworkbro.ui.me

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.AllInclusive
import androidx.compose.material.icons.outlined.CalendarMonth
import androidx.compose.material.icons.outlined.CheckCircle
import androidx.compose.material.icons.outlined.HourglassEmpty
import androidx.compose.material.icons.outlined.LocalFireDepartment
import androidx.compose.material.icons.outlined.Repeat
import androidx.compose.material.icons.outlined.TaskAlt
import androidx.compose.material.icons.outlined.Timer
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.azarai.goworkbro.R
import com.azarai.goworkbro.core.util.Dates
import com.azarai.goworkbro.ui.appString
import com.azarai.goworkbro.ui.components.AppCard
import com.azarai.goworkbro.ui.theme.AppTheme
import com.azarai.goworkbro.ui.theme.ChartColors

/** Stats tab: lifetime + today aggregates (v1 parity). */
@Composable
fun StatsTab(vm: MeViewModel) {
    val colors = AppTheme.colors
    val todos by vm.todos.collectAsState()
    val habits by vm.habits.collectAsState()
    val countdowns by vm.countdowns.collectAsState()
    val allSessions by vm.allSessions.collectAsState()
    val todaySessions by vm.todaySessions.collectAsState()
    val lifetimeTodos by vm.lifetimeTodosCompleted.collectAsState()
    val lifetimeHabits by vm.lifetimeHabitsCompleted.collectAsState()
    val firstUsed by vm.firstUsedDate.collectAsState()

    val lifetimeSeconds = allSessions.sumOf { it.durationSeconds }
    val todaySeconds = todaySessions.sumOf { it.durationSeconds }

    LazyColumn(
        contentPadding = androidx.compose.foundation.layout.PaddingValues(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp),
        modifier = Modifier.fillMaxSize(),
    ) {
        item {
            Text(
                appString(R.string.lifetime_stats),
                style = MaterialTheme.typography.titleLarge,
                color = colors.textPrimary,
            )
        }
        item {
            StatCard(appString(R.string.total_focus), formatDuration(lifetimeSeconds), Icons.Outlined.AllInclusive, ChartColors[0])
        }
        item {
            StatCard(appString(R.string.total_sessions), allSessions.size.toString(), Icons.Outlined.LocalFireDepartment, ChartColors[3])
        }
        item {
            StatCard(appString(R.string.total_todos), lifetimeTodos.toString(), Icons.Outlined.TaskAlt, ChartColors[2])
        }
        item {
            StatCard(appString(R.string.total_habits), lifetimeHabits.toString(), Icons.Outlined.Repeat, ChartColors[1])
        }
        item {
            StatCard(appString(R.string.using_since), firstUsed.take(10), Icons.Outlined.CalendarMonth, ChartColors[4])
        }
        item { Spacer(Modifier.height(12.dp)) }
        item {
            Text(
                appString(R.string.today),
                style = MaterialTheme.typography.titleLarge,
                color = colors.textPrimary,
            )
        }
        item {
            StatCard(appString(R.string.today_focus), formatDuration(todaySeconds), Icons.Outlined.Timer, ChartColors[0])
        }
        item {
            StatCard(
                appString(R.string.pomodoro_count),
                appString(R.string.count_value, todaySessions.size),
                Icons.Outlined.LocalFireDepartment,
                ChartColors[3],
            )
        }
        item {
            StatCard(
                appString(R.string.todo_completed),
                "${todos.count { it.isCompleted }} / ${todos.size}",
                Icons.Outlined.CheckCircle,
                ChartColors[2],
            )
        }
        item {
            StatCard(
                appString(R.string.habit_completed),
                "${habits.count { it.isCompleted }} / ${habits.size}",
                Icons.Outlined.Repeat,
                ChartColors[1],
            )
        }
        item {
            StatCard(
                appString(R.string.active_countdowns),
                appString(R.string.count_value, countdowns.size),
                Icons.Outlined.HourglassEmpty,
                ChartColors[4],
            )
        }
    }
}

@Composable
private fun StatCard(title: String, value: String, icon: ImageVector, color: Color) {
    val colors = AppTheme.colors
    AppCard(modifier = Modifier.fillMaxWidth()) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Box(
                modifier = Modifier
                    .size(48.dp)
                    .background(color.copy(alpha = 0.12f), RoundedCornerShape(14.dp)),
                contentAlignment = Alignment.Center,
            ) {
                Icon(icon, contentDescription = null, tint = color, modifier = Modifier.size(24.dp))
            }
            Spacer(Modifier.width(16.dp))
            Text(
                title,
                style = MaterialTheme.typography.bodyMedium,
                color = colors.textBody,
                modifier = Modifier.weight(1f),
            )
            Text(
                value,
                style = MaterialTheme.typography.titleLarge.copy(fontWeight = FontWeight.SemiBold),
                color = color,
            )
        }
    }
}

private fun formatDuration(seconds: Int): String {
    val h = seconds / 3600
    val m = (seconds % 3600) / 60
    return if (h > 0) "${h}h ${m}m" else "${m}m"
}
